test/testdrive/temporary.td

```diff
@@ -67,6 +67,19 @@ catalog item 'foo' already exists
 
 > CREATE OR REPLACE TEMPORARY MATERIALIZED VIEW foo AS SELECT * FROM v
 
+> CREATE TEMPORARY TABLE temp_t (a int, b text NOT NULL)
+
+! CREATE TEMP TABLE temp_t (a int, b text NOT NULL)
```

Comment:
The error message when mixing temporary and non-temporary objects references an internal identifier rather than the actual name of the thing. But I guess this is a more general problem not limited to temporary tables alone:

```
pstoev=> create temporary table t1 (f1 integer);
CREATE TABLE
pstoev=> create view v1 as select (select * from t1 limit 1) from t1;
ERROR:  non-temporary items cannot depend on temporary item 'u47'
```

In postgres, the view is automatically converted to a temporary one.

Reply:
Fixed the error message locally!

Reply:
I think I'll punt the conversion to a temporary object to another PR, if you don't mind! I've created an issue here: https://github.com/MaterializeInc/materialize/issues/5580.

Reply:
Yes totally please proceed as you see fit.